.github/workflows/test-unit.yml

```diff
@@ -1,6 +1,7 @@
 name: Test Unit
 
 on:
+  workflow_dispatch:
```

Comment:
please don't

Reply:
I'm just using this for code testing since I don't have complete golang environment setup, I will discard this commit when it's going to merge.

Reply:
so dont worry about that

Reply:
removed~

Reply:
thanks

Reply:
As for the complete golang environment setup, as long as you have go installed, after you rebase your PR, you should be able to call `make` localy

Reply:
hmm... I installed golang and when I call make, it generates following error:
```text
gofmt -l -w ./htransformation_test.go ./htransformation.go ./pkg/tests/require/require.go ./pkg/tests/assert/assert.go ./pkg/handler/set/set_test.go ./pkg/handler/set/set.go ./pkg/handler/deleter/delete_test.go ./pkg/handler/deleter/delete.go ./pkg/handler/join/join_test.go ./pkg/handler/join/join.go ./pkg/handler/rename/rename_test.go ./pkg/handler/rename/rename.go ./pkg/handler/rewrite/rewrite.go ./pkg/handler/rewrite/rewrite_test.go ./pkg/types/types.go
golangci-lint run
WARN [linters_context] copyloopvar: this linter is disabled because the Go version (1.19) of your project is lower than Go 1.22 
WARN [linters_context] intrange: this linter is disabled because the Go version (1.19) of your project is lower than Go 1.22 
pkg/handler/set/set.go:7:2: import 'github.com/tomMoulard/htransformation/pkg/types' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/types"
        ^
pkg/handler/deleter/delete.go:7:2: import 'github.com/tomMoulard/htransformation/pkg/types' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/types"
        ^
pkg/tests/require/require.go:6:2: import 'github.com/tomMoulard/htransformation/pkg/tests/assert' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/tests/assert"
        ^
pkg/handler/join/join.go:7:2: import 'github.com/tomMoulard/htransformation/pkg/types' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/types"
        ^
pkg/handler/rewrite/rewrite.go:29: File is not `gofumpt`-ed (gofumpt)
        req *http.Request, rw http.ResponseWriter) {
pkg/handler/deleter/delete_test.go:8:2: import 'github.com/tomMoulard/htransformation/pkg/handler/deleter' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/handler/deleter"
        ^
pkg/handler/deleter/delete_test.go:9:2: import 'github.com/tomMoulard/htransformation/pkg/tests/assert' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/tests/assert"
        ^
pkg/handler/deleter/delete_test.go:10:2: import 'github.com/tomMoulard/htransformation/pkg/tests/require' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/tests/require"
        ^
pkg/handler/deleter/delete_test.go:100:4: if statements should only be cuddled with assignments (wsl)
                        if test.wantErr {
                        ^
pkg/handler/rename/rename_test.go:9:2: import 'github.com/tomMoulard/htransformation/pkg/handler/rename' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/handler/rename"
        ^
pkg/handler/rename/rename_test.go:10:2: import 'github.com/tomMoulard/htransformation/pkg/tests/assert' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/tests/assert"
        ^
pkg/handler/rename/rename_test.go:11:2: import 'github.com/tomMoulard/htransformation/pkg/tests/require' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/tests/require"
        ^
pkg/handler/rename/rename_test.go:136:4: if statements should only be cuddled with assignments (wsl)
                        if test.wantErr {
                        ^
pkg/handler/set/set_test.go:9:2: import 'github.com/tomMoulard/htransformation/pkg/handler/set' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/handler/set"
        ^
pkg/handler/set/set_test.go:11:2: import 'github.com/tomMoulard/htransformation/pkg/tests/require' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/tests/require"
        ^
pkg/handler/set/set_test.go:135:4: if statements should only be cuddled with assignments (wsl)
                        if test.wantErr {
                        ^
htransformation.go:9:2: import 'github.com/tomMoulard/htransformation/pkg/handler/deleter' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/handler/deleter"
        ^
htransformation.go:10:2: import 'github.com/tomMoulard/htransformation/pkg/handler/join' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/handler/join"
        ^
htransformation.go:11:2: import 'github.com/tomMoulard/htransformation/pkg/handler/rename' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/handler/rename"
        ^
htransformation.go:12:2: import 'github.com/tomMoulard/htransformation/pkg/handler/rewrite' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/handler/rewrite"
        ^
htransformation.go:13:2: import 'github.com/tomMoulard/htransformation/pkg/handler/set' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/handler/set"
        ^
htransformation_test.go:168:34: unused-parameter: parameter 'rw' seems to be unused, consider removing or renaming it as _ (revive)
                        next := http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {})
                                                      ^
htransformation_test.go:9:2: import 'github.com/tomMoulard/htransformation' is not allowed from list 'Main' (depguard)
        plug "github.com/tomMoulard/htransformation"
        ^
pkg/handler/join/join_test.go:8:2: import 'github.com/tomMoulard/htransformation/pkg/handler/join' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/handler/join"
        ^
pkg/handler/rewrite/rewrite_test.go:9:2: import 'github.com/tomMoulard/htransformation/pkg/handler/rewrite' is not allowed from list 'Main' (depguard)
        "github.com/tomMoulard/htransformation/pkg/handler/rewrite"
        ^
make: *** [Makefile:10: lint] Error 1
```
Maybe I should degrade golang to 1.20?

Reply:
nope, you just need to update your branch with the latest fixes I introduced in #60 (i.e. `git pull upstream main --rebase`).